tests/python/pants_test/engine/test_test.py

```
@@ -0,0 +1,72 @@
+
+from future.utils import text_type
+
+from pants.build_graph.address import Address
+from pants.engine.console import Console
+from pants.engine.legacy.graph import HydratedTarget
+from pants.engine.rules import console_rule
+from pants.engine.selectors import Get, Select
+from pants.util.meta import AbstractClass
+from pants.util.objects import datatype
+from pants_test.engine.scheduler_test_base import SchedulerTestBase
+from pants_test.engine.util import MockConsole, run_rule
+from pants_test.test_base import TestBase
+
+
+# TODO: Move to production code
+class Status(object):
+  SUCCESS = "SUCCESS"
+  FAILURE = "FAILURE"
+
+
+# TODO: Pretty sure these shouldn't be text_types.
+# status should be an enum somehow, and stdout should be...
+# I guess text_type, actually... But also maybe bytes?
+# When do we start asserting that testrunner output is unicode?
+class TestResult(datatype([
+  # One of the Status pseudo-enum values capturing whether the run was successful.
+  ('status', text_type),
+  # The stdout of the test runner (which may or may not include actual testcase output).
+  ('stdout', text_type)
+])):
+  pass
+
+
+# TODO: Move to production code
+@console_rule('test', [Select(Console), Select(TestResult)])
+def fast_test(console, hydrated_targets):
+  test_results = yield [Get(TestResult, HydratedTarget, ht) for ht in hydrated_targets]
+  wrote_any_stdout = False
+  for test_result in test_results:
+    wrote_any_stdout |= bool(test_result.stdout)
+    # Assume \n-terminated
+    console.write_stdout(test_result.stdout)
+    if test_result.stdout and not test_result.stdout[-1] == '\n':
+      console.write_stdout('\n')
+  if wrote_any_stdout:
+    console.write_stdout('\n')
+  for (hydrated_target, test_result) in zip(hydrated_targets, test_results):
+    console.print_stdout('{0:80}.....{1:>10}'.format(hydrated_target.address.reference(), test_result.status))
+  # TODO: Communicate an exit code out of the console_rule (if tests failed)
```

This rule should raise an exception for failures.